tests/endtoend/reverse/reverse_modit.py

```diff
@@ -4,149 +4,149 @@
 
-#settings before HMC
+# settings before HMC
 vsini_max = 100.0
 vr_array = velocity_grid(res, vsini_max)
 
+
 # a core driver
 def frun(Tarr, MMR_CH4, Mp, Rp, u1, u2, RV, vsini):
-    g = 2478.57730044555*Mp/Rp**2
-    SijM_CH4, ngammaLM_CH4, nsigmaDl_CH4 = modit.exomol(
-        mdbCH4, Tarr, Parr, R, molmassCH4)
-    xsm_CH4 = modit.xsmatrix(
-        cnu, indexnu, R, pmarray, nsigmaDl_CH4, ngammaLM_CH4, SijM_CH4, nus, dgm_ngammaL)
-    # abs is used to remove negative values in xsv
-    dtaumCH4 = dtauM(dParr, jnp.abs(xsm_CH4), MMR_CH4*ONEARR, molmassCH4, g)
+    gravity = gravity_jupiter(Rp=Rp, Mp=Mp)  # gravity in the unit of Jupiter
+    # CH4
+    xsmatrix = opa.xsmatrix(Tarr, art.pressure)
+    mmr_profile = art.constant_mmr_profile(MMR_CH4)
+    dtaumCH4 = art.opacity_profile_lines(xsmatrix, mmr_profile,
+                                         opa.mdb.molmass, gravity)
     # CIA
-    dtaucH2H2 = dtauCIA(nus, Tarr, Parr, dParr, vmrH2, vmrH2,
-                        mmw, g, cdbH2H2.nucia, cdbH2H2.tcia, cdbH2H2.logac)
-    dtau = dtaumCH4+dtaucH2H2
-    sourcef = planck.piBarr(Tarr, nus)
-    F0 = rtrun(dtau, sourcef)/norm
+    logacia = opcia.logacia_matrix(Tarr)
+    dtaucH2H2 = art.opacity_profile_cia(logacia, Tarr, vmrH2, vmrH2, mmw,
+                                        gravity)
+
+    # RT
+    dtau = dtaumCH4 + dtaucH2H2
+    F0 = art.run(dtau, Tarr) / norm
     Frot = convolve_rigid_rotation(F0, vr_array, vsini, u1, u2)
-    mu = ipgauss_sampling(nusd, nus, Frot, beta_inst, RV)
+    mu = ipgauss_sampling(nusd, nu_grid, Frot, beta_inst, RV, vr_array)
     return mu
 
 
-# test
-if False:
-    Tarr = 1200.0*(Parr/Pref)**0.1
-    mu = frun(Tarr, MMR_CH4=0.0059, Mp=33.2, Rp=0.88,
-              u1=0.0, u2=0.0, RV=10.0, vsini=20.0)
-    plt.plot(wavd, mu)
-    plt.show()
-
-    
-
-
-def model_c(nu1, y1):
+def model_c(y1):
     Rp = numpyro.sample('Rp', dist.Uniform(0.4, 1.2))
     RV = numpyro.sample('RV', dist.Uniform(5.0, 15.0))
     MMR_CH4 = numpyro.sample('MMR_CH4', dist.Uniform(0.0, 0.015))
     T0 = numpyro.sample('T0', dist.Uniform(1000.0, 1500.0))
     alpha = numpyro.sample('alpha', dist.Uniform(0.05, 0.2))
     vsini = numpyro.sample('vsini', dist.Uniform(15.0, 25.0))
-    g = 2478.57730044555*Mp/Rp**2  # gravity
     u1 = 0.0
     u2 = 0.0
     # T-P model//
-    Tarr = T0*(Parr/Pref)**alpha
-    # line computation CH4
+    Tarr = art.powerlaw_temperature(T0, alpha)
     mu = frun(Tarr, MMR_CH4, Mp, Rp, u1, u2, RV, vsini)
     numpyro.sample('y1', dist.Normal(mu, sigmain), obs=y1)
 
 
 rng_key = random.PRNGKey(0)
 rng_key, rng_key_ = random.split(rng_key)
 num_warmup, num_samples = 300, 600
-#kernel = NUTS(model_c, forward_mode_differentiation=True)
+# kernel = NUTS(model_c, forward_mode_differentiation=True)
 kernel = NUTS(model_c, forward_mode_differentiation=False)
 
 mcmc = MCMC(kernel, num_warmup=num_warmup, num_samples=num_samples)
-mcmc.run(rng_key_, nu1=nusd, y1=nflux)
+mcmc.run(rng_key_, y1=nflux)
```

Comment:
Also in line 154, nu1 should be removed as `predictions = pred(rng_key_, y1=None)` although I'm afraid a very minor correction.

Reply:
thanks!

Reply:
corrected.